students/aravichander/lesson06/Mailroom3.py

```diff
@@ -30,60 +32,70 @@ def letterofthanks(donorname,totaldonated):
 	print("\nLetter of thanks has been saved to hard disk as text file under "+str(filename))
 
 def searchdonors(thankdonor):
-	#Not sure if these exceptions here are needed to be honest#
-	for x in Donors:
 		try:
-			if thankdonor == x:
-				totaldonated = sum(Donors[x])
-				letterofthanks(thankdonor,totaldonated)
+			print(Donors[thankdonor])
```

Comment:
as above -- you don't want your searching function to print anything -- it should return the donor if found, and either None or raise an exception is not found.